app/services/intercom_service.rb

```diff
@@ -129,11 +129,22 @@ def self.intercom
     @intercom ||= Intercom::Client.new(token: EnvironmentCredentials.dig(:intercom, :intercom_access_token))
   end
 
+  MAX_RETRY_COUNT = 3
+
   def self.intercom_api(collection, verb, params)
     if Rails.env.development?
       Rails.logger.debug("Calling Intercom: #{collection}.#{verb}(#{params})")
     end
-    result = intercom.send(collection).send(verb, params)
+
+    retry_counts = 0
+
+    begin
+      result = intercom.send(collection).send(verb, params)
+    rescue
+      retry_counts += 1
+      retry if retry_counts < MAX_RETRY_COUNT
```

Comment:
This may only retry twice since we increment retry_counts before checking if it's less than 3.

We could possibly change `< MAX_RETRY_COUNT` to `<= MAX_RETRY_COUNT`, but there may be reasons why we only want a max of 2 retries.